query.lua

```diff
@@ -0,0 +1,50 @@
+function safe_pairs(item, keys_only)
+    if keys_only then
+        local mt = debug.getmetatable(item)
+        if mt and mt._index_table then
+            local idx = 0
+            return function()
+                idx = idx + 1
+                if mt._index_table[idx] then
+                    return mt._index_table[idx]
+                end
+            end
+        end
+    end
+    local ret = table.pack(pcall(function() return pairs(item) end))
+    local ok = ret[1]
+    table.remove(ret, 1)
+    if ok then
+        return table.unpack(ret)
+    else
+        return function() end
+    end
+end
+
+function Query(table, query, parent)
+    if not parent then
+        parent = ""
+    end
+    for k,v in safe_pairs(table) do
+        if not tonumber(k) and type(k) ~= "table" and not string.find(tostring(k), 'script') then

```

Comment:
This line should probably have a comment on it explaining why these checks are made. The reason is essentially to prevent infinite recursion.